### DISCUSSION TITLE
i18next::backendConnector fails to parse url

### DISCUSSION THREAD

**Discussion by AndonMitev** [i18next/next-i18next#2247] [General]
Hello, i do have following structure:

<img width="298" alt="image" src="https://github.com/i18next/next-i18next/assets/32274987/65dfa204-ba92-4846-a9db-4ea24270af5d">

and my i18n config looks like:

```
import i18n from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import Backend from 'i18next-http-backend';
import { initReactI18next } from 'react-i18next';

if (!i18n.isInitialized) {
  i18n.use(Backend).use(LanguageDetector).use(initReactI18next).init({
    lng: 'ja',
    fallbackLng: 'ja',
    debug: true,
  });
}

export default i18n;

```

The error that i'm facing tho is:
`i18next::backendConnector: loading namespace translation for language ja failed TypeError: Failed to parse URL from /locales/ja/translation.json`

No idea why is happening is it some race condition or so. JSON file is valid also checked network tab

<img width="631" alt="image" src="https://github.com/i18next/next-i18next/assets/32274987/4e9c5095-9c51-4eb0-8108-9890fb1be4b5">

using `nextjs@13.4.13` if that matter. 

Additional: I've started noticing this behaviour when i had to translate constant files and not react components. They are translated successfully but still manage to see this error:

<img width="1181" alt="image" src="https://github.com/i18next/next-i18next/assets/32274987/61ba51c4-b76d-48e7-a0af-cef809bf0b74">

I can see that is trying to load the language before being initialized but i assume that expected in order to be iniitialized. This is how translations are loaded in constant files:

```
import i18n from "../i18n";

i18n.on("loaded", () => {
  updateSec1TextContent();
})

const updateSec1TextContent = () => {
  sec1.title = i18n.t("terms.sec1.text1");
  sec1.content = [
    { subSectionNumber: 1, text: i18n.t("terms.sec1.content.text1") },
    { subSectionNumber: 2, text: i18n.t("terms.sec1.content.text2") },
  ];
};
```
i dont think there is something suspicious here but no idea why this initial loading is failing :X

**Comment by adrai** [comment]
You should not initialize i18next yourself in the config file, if you use next-i18next... have a look at: https://github.com/i18next/i18next-http-backend/blob/master/example/next/next-i18next.config.js

**Reply by AndonMitev** [reply]
@adrai thanks check it, but how you are going to import `i18next.t` to use it in `constant` files since custom hooks are not applicable there

**Reply by adrai** [reply]
since i18next init is async (if not using [inline translation resources](https://www.i18next.com/how-to/add-or-load-translations#add-on-init)) there is no way to call the t function outside of an async loop (for example in a constant file)

**Reply by adrai** [reply]
https://github.com/i18next/react-i18next/issues/1236#issuecomment-1872246879
https://github.com/i18next/react-i18next/issues/909#issuecomment-1871099705

**Reply by AndonMitev** [reply]
Yeah currently i'm using exact this mechanism:

https://github.com/i18next/react-i18next/issues/1236#issuecomment-762039023

but if i switch avoiding initializing `i18next` and there is no instance how i can import `i18next` and subscribe to these async events, because the first example provided here is by adding i18n to `next.config.json` and i18n contains only the config it self but no instance, any possible way to use instance from `next.config.json`?

**Reply by adrai** [reply]
that's not really supported with next-i18next

**Reply by AndonMitev** [reply]
hmm but isn't that suggested approach from the linked issued https://github.com/i18next/react-i18next/issues/1236#issuecomment-762039023 here they are doing something similar if i'm not mistaken

**Reply by adrai** [reply]
That's not for next-i18next

**Reply by AndonMitev** [reply]
so what is your suggestion, provide inline translation resources and using `t` function in constant files somehow?

**Reply by adrai** [reply]
The question is why do you need constant translation at all?
Normally for next.js the translations should depend on an incoming request (for example to detect the correct language etc.)
So maybe first explain (or show in a real example) what you want to achieve and why...

**Reply by adrai** [reply]
fyi:
https://github.com/i18next/next-i18next/issues/2079#issuecomment-1386970151
https://github.com/i18next/next-i18next/issues/2171

**Reply by AndonMitev** [reply]
Ok. thanks to you comment i did some adjustments and now is working:

Changes:
1. Languages are imported locally:
```
  en: {
          translation: {
            ...en,
          },
        },
        ja: {
          translation: {
            ...ja,
          },
        },
```

2. Importing `import { t } from 'i18next'` instead of the created instance seems to work

**Comment by AndonMitev** [comment]
Thanks @adrai !
